src/System/Handleable.h

```diff
@@ -0,0 +1,44 @@
+#ifndef _HANDLEABLE_H
+#define _HANDLEABLE_H
+
+#include "Particle.h"
+#include "Handler.h"
+// Forward declaration due to mutual inclusion
+class Handler;
+
+// Abstract class for any object that can be "handleable"
+// (ie. would need a function to run regularly in the loop)
+class Handleable {
+    public:
+
+        // One handler is shared by all handleables
+        static Handler handler;
```

Comment:
when do static members get constructed?  Like what would happen if before any Handleables had been initialized, we tried to access Handleable::handler?

Reply:
That's what I was struggling with! The handler is constructed in a .cpp file. See the top of main.cpp, I'm constructing it there so that it happens first before any Handleables are created.

Reply:
I think making the Handler class a singleton, or creating a singleton member handler would make a lot of sense in this case.  Might be good to practice using the singleton pattern too

Reply:
I just implemented it following the instance() example from PublishQueuePosixRK and it works!  pretty small change, and it makes it so you don't have to initialize the handler at the top of main.  Initialization would happen wherever instance() gets called first: Handler::instance() or Handleable::handlerInstance() (I created a Singleton member handler in Handleable)

Reply:
in Handleable.h:

```
public:
    static Handler &handlerInstance();
    ...
private:
    static Handler* _handlerInstance;
```

in Handleable.cpp
```
Handler* Handleable::_handlerInstance;

Handler &Handleable::handlerInstance() {
    if (!_handlerInstance) {
        _handlerInstance = new Handler();
    }
    return *_handlerInstance;
}
```

then anywhere you want to use handleable's handler:

```
Handleable::handlerInstance().handle();
...
```

Reply:
Just realized, if you want to keep the code the same but avoid initializing Handleable::handler in main, you can do it at the top of the Handleable.cpp file

Reply:
Great idea, I'll try your singleton suggestion. As for simply initializing the handler at the top of Handleable.cpp, that unfortunately didn't work. It would construct a bunch of the Handleables at the top of main.cpp before getting to Handleable.cpp, leading to a segfault.

Reply:
Oh strange, I compiled it but didn't try running it on the boron.  I've tested the singleton implementation and it works fine though.  Do you think making the Handler a singleton class makes more sense?  I can't think of a reason to ever have more than one group of handleables

Reply:
I can't think of a reason either, up to you if you think we should do that. I've made the changes you suggested in the latest commit.

Reply:
The 'instance' getter is supposed to encapsulate construction.  Delegating construction to the Handleable::add()  means that our program will break if no handleables have been added before setup/loop are called because Handleable::handlerInstance() will return null.  Either way I think it probably makes the most sense to turn the Handler into a singleton class.  I can make those changes

Reply:
just went and did that.  seems fine to merge unless there's anything you want to change ^_^

Reply:
Looks good! If you've tested it and it all works, feel free to merge it.